src/components/CategoryPicker/index.js

```diff
@@ -1,47 +1,91 @@
-import OptionsList from '../OptionsList';
 import styles from '../../styles/styles';
-import ScreenWrapper from '../ScreenWrapper';
 import Navigation from '../../libs/Navigation/Navigation';
 import ROUTES from '../../ROUTES';
+import CONST from '../../CONST';
+import * as IOU from '../../libs/actions/IOU';
+import * as OptionsListUtils from '../../libs/OptionsListUtils';
+import OptionsSelector from '../OptionsSelector';
+import useLocalize from '../../hooks/useLocalize';
+
+function CategoryPicker({policyCategories, reportID, iouType, iou, policyRecentlyUsedCategories}) {
+    const {translate} = useLocalize();
+    const [searchValue, setSearchValue] = useState('');
+
+    const policyCategoriesCount = _.size(policyCategories);
+    const isCategoriesCountBelowThreshold = policyCategoriesCount < CONST.CATEGORY_LIST_THRESHOLD;
 
-function CategoryPicker({policyCategories, reportID, iouType}) {
-    const sections = useMemo(() => {
-        const categoryList = _.chain(policyCategories)
-            .values()
-            .map((category) => ({
-                text: category.name,
-                keyForList: category.name,
-                tooltipText: category.name,
-            }))
-            .value();
+    const selectedOptions = useMemo(() => {
+        if (!iou.category) {
+            return [];
+        }
 
         return [
             {
-                data: categoryList,
+                name: iou.category,
+                enabled: true,
+                accountID: null,
             },
         ];
-    }, [policyCategories]);
+    }, [iou.category]);
+
+    const initialFocusedIndex = useMemo(() => {
+        if (isCategoriesCountBelowThreshold && selectedOptions.length > 0) {
+            return _.chain(policyCategories)
+                .values()
+                .findIndex((category) => category.name === selectedOptions[0].name, true)
+                .value();
+        }
+
+        return 0;
+    }, [policyCategories, selectedOptions, isCategoriesCountBelowThreshold]);
+
+    const sections = useMemo(
+        () => OptionsListUtils.getNewChatOptions({}, {}, [], searchValue, selectedOptions, [], false, false, true, policyCategories, policyRecentlyUsedCategories, false).categoryOptions,
+        [policyCategories, policyRecentlyUsedCategories, searchValue, selectedOptions],
+    );
+
+    const headerMessage = OptionsListUtils.getHeaderMessage(lodashGet(sections, '[0].data.length', 0) > 0, false, searchValue);
+    const shouldShowTextInput = !isCategoriesCountBelowThreshold;
 
     const navigateBack = () => {
         Navigation.goBack(ROUTES.getMoneyRequestConfirmationRoute(iouType, reportID));
     };
 
+    const updateCategory = (category) => {
+        if (category.searchText === iou.category) {
+            IOU.resetMoneyRequestCategory();
+        } else {
+            IOU.setMoneyRequestCategory({
+                name: category.searchText,
+                enabled: true,
+            });
```

Comment:
Let's simplify, since we use only name of a category.
```suggestion
            IOU.setMoneyRequestCategory(category.searchText);
```